src/code/code_80140E80.c

```diff
@@ -1,21 +1,359 @@
+/**
+ * @file code_80140E80.c
+ * @brief Copies images between color images (generally framebuffers), possibly with filling, scaling, and colour
+ * interpolation.
+ *
+ * Used for several transition effects, and in z_play to shrink the screen at the end of the First and Second Days.
+ *
+ *
+ *
+ * @note to use the functions in this file, with the exception of func_80141778(), it is necessary to load the S2DEX2
+ * microcode first, and then re-load the 3D microcode afterwards for the rest of the drawing in the frame; see
+ * func_80141778() for how to do this.
+ */
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_80140E80/func_80140E80.s")
+// z64.h once named properly
+typedef enum {
+    /* 0 */ BG_CYC_1CYC, // Enable rescaling
+    /* 1 */ BG_CYC_COPY  // Direct copy, no scaling
+} BgMode;
```

Comment:
```suggestion
    /* 0 */ BG_CYCLE_1CYC, // Enable rescaling
    /* 1 */ BG_CYCLE_COPY  // Direct copy, no scaling
} BgMode;
```
Regardless of whatever this is named to later, I think we can afford the extra letters

Reply:
CYC is the standard abbreviation gbi.h uses:
```

/* G_SETOTHERMODE_H gSetCycleType */
#define	G_CYC_1CYCLE		(0 << G_MDSFT_CYCLETYPE)
#define	G_CYC_2CYCLE		(1 << G_MDSFT_CYCLETYPE)
#define	G_CYC_COPY		(2 << G_MDSFT_CYCLETYPE)
#define	G_CYC_FILL		(3 << G_MDSFT_CYCLETYPE)```

Reply:
I guess we could just use that gbi does, at least for now